pkg/ddl/partition.go

```diff
@@ -1399,21 +1417,22 @@ func buildRangePartitionDefinitions(ctx sessionctx.Context, defs []*ast.Partitio
 				return nil, exprChecker.err
 			}
 			// If multi-column use new evaluated+normalized output, instead of just formatted expression
-			if len(partValStrings) > i && len(colTypes) > 1 {
+			if len(partValStrings) > i && len(tbInfo.Partition.Columns) > 0 {
 				partVal := partValStrings[i]
-				switch colTypes[i].EvalType() {
-				case types.ETInt:
-					// no wrapping
-				case types.ETDatetime, types.ETString, types.ETDuration:
-					if _, ok := clause.Exprs[i].(*ast.MaxValueExpr); !ok {
-						// Don't wrap MAXVALUE
-						partVal = driver.WrapInSingleQuotes(partVal)
+				if _, ok := clause.Exprs[i].(*ast.MaxValueExpr); !ok {
+					partVal, err = generatePartValuesWithTp(partVal, colTypes[i])
+					if err != nil {
+						return nil, err
 					}
-				default:
-					return nil, dbterror.ErrWrongTypeColumnValue.GenWithStackByArgs()
 				}
+
 				piDef.LessThan = append(piDef.LessThan, partVal)
 			} else {
+				// TODO: return the integer value from `checkPartitionValuesIsInt` and use it to construct `piDef` like
+				// the above condition.
+				// It's fine to not consider the collation/charset problem as it only supports integer now, but once it
+				// supports more functions and other types, we'll need to reconsider how to format the expression in this
+				// situation.
```

Comment:
I don't think RANGE/LIST/HASH will support other types than INT based ones, that is the reason for RANGE/LIST COLUMNS. IIRC adding COLUMNS partitioning was to separate that implementation from the simpler INT based ones when it was added in MySQL.

Reply:
Got it. Thanks for the clarifications.